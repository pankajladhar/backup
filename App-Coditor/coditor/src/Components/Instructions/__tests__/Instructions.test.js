import React from "react";
import renderer from "react-test-renderer";
import Instructions from "./../Instructions";

describe("<Instructions /> ", () => {
  const props = {
    data: {
      title: "title",
      description: "description"
    }
  };

  it("should render properly", () => {
    const tree = renderer.create(<Instructions {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should render properly when notes are present", () => {
    const updatedProps = JSON.parse(JSON.stringify(props));
    updatedProps.data.notes = "some notes";
    const tree = renderer.create(<Instructions {...updatedProps} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should render properly when exmaples are present", () => {
    const updatedProps = JSON.parse(JSON.stringify(props));
    updatedProps.data.examples = [
      {
        functionExample: "functionExample",
        expectedOutput: "expectedOutput",
        comment: "comment"
      }
    ];
    const tree = renderer.create(<Instructions {...updatedProps} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
