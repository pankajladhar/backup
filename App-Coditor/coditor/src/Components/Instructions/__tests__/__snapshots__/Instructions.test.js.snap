// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Instructions />  should render properly 1`] = `
<div
  className="p-12"
>
  <h3
    className="text-3xl font-headline uppercase"
  >
    title
  </h3>
  <p
    className="text-gray-600"
  >
    description
  </p>
  <ul />
</div>
`;

exports[`<Instructions />  should render properly when exmaples are present 1`] = `
<div
  className="p-12"
>
  <h3
    className="text-3xl font-headline uppercase"
  >
    title
  </h3>
  <p
    className="text-gray-600"
  >
    description
  </p>
  <div
    className="my-4 text-lg"
  >
    Examples
  </div>
  <ul>
    <li
      className="flex mb-1"
    >
      <span
        className="bg-gray-300 py-1 px-2 rounded mr-3 text-gray-700"
      >
        functionExample
      </span>
      <span
        className="mr-3 text-gray-500"
      >
         =&gt; 
      </span>
      <span
        className="mr-3 font-bold"
      >
        expectedOutput
      </span>
      <span
        className="text-gray-500 italic"
      >
        comment
      </span>
    </li>
  </ul>
</div>
`;

exports[`<Instructions />  should render properly when notes are present 1`] = `
<div
  className="p-12"
>
  <h3
    className="text-3xl font-headline uppercase"
  >
    title
  </h3>
  <p
    className="text-gray-600"
  >
    description
  </p>
  <ul />
  <div
    className="my-4 text-lg"
  >
    Notes
  </div>
  <p
    className="text-gray-600"
  >
    some notes
  </p>
</div>
`;
